ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,515 @@
+
+        variable_data = model_data[variable_name]
+        # Ensure the data is in the correct dimension order
+        if variable_data.dims != ('latitude', 'longitude', 'time'):
+            variable_data = variable_data.transpose('latitude', 'longitude', 'time')
+
+        model_grid = (variable_data.coords['latitude'].values,
+                      variable_data.coords['longitude'].values,
+                      valid_times_unix)
+
+        interpolator = RegularGridInterpolator(model_grid, variable_data.values, bounds_error=False, fill_value=np.nan)
+        interpolated_values = interpolator(valid_satellite_points)
+
+        # Handle model times for interpolated values based on satellite times
+        model_times_for_interpolated_values = np.searchsorted(valid_times_unix, satellite_times_unix[mask], side='right') - 1
+        model_times_for_interpolated_values = valid_times_unix[model_times_for_interpolated_values]
+
+        # Apply NaN where interpolated values are NaN
+        nan_indices = np.isnan(interpolated_values)
+        model_times_for_interpolated_values[nan_indices] = np.nan
+
+        full_interpolated_values = np.full(len(satellite_times_unix), np.nan)
+        full_interpolated_values[mask] = interpolated_values
+
+        full_model_times = np.full(len(satellite_times_unix), np.nan)
+        full_model_times[mask] = model_times_for_interpolated_values
+
+        return full_interpolated_values, full_model_times
+
+    # Interpolate for each variable
+    interpolated_htsgw_surface_values, interpolated_htsgw_surface_times = prepare_and_interpolate(
+        model_data_combined, 'HTSGW_surface', filtered_satellite_latitudes, filtered_satellite_longitudes, filtered_satellite_times_unix, model_time_seconds)
+
+    interpolated_wind_surface_values, interpolated_wind_surface_times = prepare_and_interpolate(
+        model_data_combined, 'WIND_surface', filtered_satellite_latitudes, filtered_satellite_longitudes, filtered_satellite_times_unix, model_time_seconds)
+
+    fcst_hr2_ws = np.full_like(interpolated_wind_surface_values, np.nan, dtype='float64')
+   # fcst_hr2_ws = (interpolated_wind_surface_times - model_reference_time_unix) / 3600
+    valid_indices = ~np.isnan(interpolated_wind_surface_values)
+    fcst_hr2_ws[valid_indices] = (filtered_satellite_times_unix[valid_indices] - model_reference_time_unix) / 3600 
+    fcst_hr = np.full_like(filtered_satellite_times_unix, np.nan, dtype='float64')
+
+    for i, value in enumerate(interpolated_wind_surface_values):
+        if not np.isnan(value):
+            fcst_hr[i] = filtered_satellite_times_unix[i]
+
+    reference_time, reference_date = extract_reference_times(model_data_files[0])
+
+    # Extract hs and wsp from satellite data, filtered by time
+    hs = satellite_data['hs'].sel(time=satellite_data['time'].values[satellite_times_unix <= max_model_time])
```

Comment:
For the grib2 section the following is defined: 
`  time_mask = (satellite_times_unix >= min_model_time) & (satellite_times_unix <= max_model_time)`

Is there a reason that netcdf only does part of this masking?

Reply:
Addressed.

Reply:
@Ghazal-Mohammadpour can you expand a bit more here on how this was addressed?  It seems to me that the netcdf and grib sections still mask out time differently, but perhaps I'm overlooking something?